engine/engine_test.go

```diff
@@ -3150,29 +3150,68 @@ func TestNativeHistogram(t *testing.T) {
 	}
 
 	cases := []struct {
-		name  string
-		query string
+		name                   string
+		query                  string
+		wantEmptyForMixedTypes bool
 	}{
 		{
 			name:  "plain selector",
 			query: "native_histogram_series",
 		},
 		{
-			name:  "irate() with native histogram",
+			name:  "irate()",
 			query: "rate(native_histogram_series[1m])",
 		},
 		{
-			name:  "rate() with native histogram",
+			name:  "rate()",
 			query: "rate(native_histogram_series[1m])",
 		},
 		{
-			name:  "increase() with native histogram",
+			name:  "increase()",
 			query: "increase(native_histogram_series[1m])",
 		},
 		{
-			name:  "delta() with native and counter histogram",
+			name:  "delta()",
 			query: "delta(native_histogram_series[1m])",
 		},
+		{
+			name:                   "sum()",
+			query:                  "sum(native_histogram_series)",
+			wantEmptyForMixedTypes: true,
+		},
+		{
+			name:                   "sum by (foo)",
+			query:                  "sum by (foo) (native_histogram_series)",
+			wantEmptyForMixedTypes: true,
+		},
+		{
+			name:  "count",
+			query: "count (native_histogram_series)",
+		},
+		{
+			name:  "count by (foo)",
+			query: "count by (foo) (native_histogram_series)",
+		},
+		// TODO(fpetkovski): The Prometheus engine returns an incorrect result for this case.
+		// Uncomment once it gets fixed.
```

Comment:
Maybe there's some Prometheus issue for this? We could link it here.

Reply:
Good point, let me reach out in the native-histograms-dev slack channel and see what the consensus is on these edge cases.